13 feedback

NathanSegerlind opened the pull request:
This implements the basic "feedback" functionality that takes the records from flow_scores.csv and for those entries marked as a 3,  those words are duplicated (thereby making such words less anomalous...)

NathanSegerlind committed:
13_feedback

Picking up feedback file and duplicating records for connections that checkout as non-suspicious.

NathanSegerlind committed:
13_feedback

Implementation of elementary feedback procedure:
- if a feedback file is in LPATH/flow_scores.csv,
--  those entries tagged with a 3 are duplicated before being inserted into the corpus

grantbabb commented on flow_pre_lda.scala:
```diff
@@ -134,18 +142,139 @@ def bin_column(value: String, cuts: Array[Double]) = {
 }
 
 
+def convert_feedback_row_to_flow_row(feedBackRow: String) = {
```
I would definitely prefer that we use the column name approach that is used in DNS pipeline over this.  It would reduce code and also be more robust to changes in column order

NathanSegerlind replied:
ohmygodyes

...

but I didn't want to do it in this checkin, seemed like too much to cram into a hotfix kinda situation

grantbabb commented on flow_pre_lda.scala:
```diff
@@ -134,18 +142,139 @@ def bin_column(value: String, cuts: Array[Double]) = {
 
-val datanoheader = removeHeader(rawdata)
-val datagood = datanoheader.filter(line => line.split(",").length == 27)
+    val nonThreateningRows = Source.fromFile(scoredFile).getLines().toArray.drop(1).filter(l=>(l.split(',')(0).toInt == 3))
```
can we pick a more neutral name here?  the code is intended to be generic and named in terms of what it does, not the domain itself

NathanSegerlind replied:
fair enough... i just grabbed the name out of my hat...

rowsToDuplicate?

grantbabb replied:
that sounds good.

grantbabb commented on flow_pre_lda.scala:
```diff
@@ -134,18 +142,139 @@ def bin_column(value: String, cuts: Array[Double]) = {
+    val nonThreateningRows = Source.fromFile(scoredFile).getLines().toArray.drop(1).filter(l=>(l.split(',')(0).toInt == 3))
+    println("User feedback read from: " + scoredFile + ". "
+      + nonThreateningRows.length + " many connections flagged nonthreatening.")
+    println("Duplification factor: " + duplicationFactor)
```
Should be "Duplication factor:"

NathanSegerlind replied:
says you and dictionary.com

(I'll change it)

grantbabb commented on ml_ops.sh:
```diff
@@ -20,6 +20,7 @@ source /etc/duxbay.conf
 export DPATH
 export HPATH
 export LUSER
+export LPATH
```
Good catch

grantbabb commented:
Reviewed with comments inline on code files.
Over all, it would be better to use one of two existing approaches for handling columns before creating a third one.  Other than that, just some minor variable name and spelling comments.

grantbabb commented:
All answers are satisfactory.  If tests pass, looks ready to merge.

NathanSegerlind committed:
13_feedback

incorporated code review comments

NathanSegerlind committed:
Merge branch '1.0.1' into 13_feedback

NathanSegerlind committed:
Merge remote-tracking branch 'upstream/1.0.1' into 13_feedback

NathanSegerlind committed:
13_feedback

made processing of flow_results.csv more robust so it can handle malformed lines without barfing, in particular, trailing newlines in the file

NathanSegerlind commented:
one last test with integration w/ latest:  1.0.1

functionally verified that modifying the flow_scores.csv changes the flow_result.csv output as expected
